lib/net/ldap/auth_adapters/gss_spnego.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/ldap/auth_adapter'
+require 'net/ldap/auth_adapters/sasl'
+
+module Net
+  class LDAP
+    module AuthAdapers
+      #--
+      # PROVISIONAL, only for testing SASL implementations. DON'T USE THIS YET.
+      # Uses Kohei Kajimoto's Ruby/NTLM. We have to find a clean way to
+      # integrate it without introducing an external dependency.
+      #
+      # This authentication method is accessed by calling #bind with a :method
+      # parameter of :gss_spnego. It requires :username and :password
+      # attributes, just like the :simple authentication method. It performs a
+      # GSS-SPNEGO authentication with the server, which is presumed to be a
+      # Microsoft Active Directory.
+      #++
+      class GSS_SPNEGO < Net::LDAP::AuthAdapter
+        def bind(auth)
+          require 'ntlm'
```

Comment:
This adapter depends on `ntlm` as the above comments point out.  now this can be defined as a gem like `net-ldap-auth_adapters-gss_spnego`.